### DISCUSSION TITLE
Best way to run with unsupported meteorological forcing products?

### DISCUSSION THREAD

**Discussion by notstarboard** [StormSurgeLive/asgs#1145] [Q&A]
Hi folks! We noticed some pretty inaccurate wind fields the NHC-advisory-driven GAHM runs I did in advance of the upcoming RICHAMP tabletop exercise. GFS actually did quite well, so we may end up switching over to GFS for future runs, but it naturally re-opens the question of the best forcing to use for our RICHAMP forecasts.

If we wanted to run ASGS with another unsupported forcing source, is there a way you'd suggest going about that? My thought was that we could run in GFS-forced mode, but hard-code a path for the forcing file and drop the file of our choice there; the GFS file should just be an ASCII file, and I recall Jason saying that local files are used over remote files if available (I think this was when I was asking about running a hindcast with GFS forcing). We could use a hook script to somehow acquire the file we need and move it to the configured folder; that could be a MetGet call or something else.

Would there be another better way of doing this, or does this sound reasonable? Thanks!

**Comment by wwlwpd** [comment]
When you say "unsupported" do you mean the data source or does this require some additional fort.15 tweaks (and/or additional processing or supporting fort.X files) also?

I don't know much about MetGet - a while back we started a basic client but it was just a demo and in a different repo. I have no plans to further develop it unless we need it ourselves. But I do recall a 2 part asynchronous approach to a) tell MG to "get" the data, then b) poll until it's available seemed to overcomplicate how we use meteorological forcing sources operationally, which is in "blocking" way (it's expected to already be there); for NHC we do poll and wait for the next forecast.

**Reply by notstarboard** [reply]
Ah! I just mean it's something that you couldn't run in real time via simple config file tweaks (e.g. TROPICALCYCLONE=on, or BACKGROUNDMET=GFS). As far as I know you can only do that for NAM, GFS, and NHC advisories (which are used with GAHM), since those are the only sources ASGS will actually download for you. 

I guess my question is: "Let's say we have a black box method for generating or downloading a wind forcing file in real time in the same format ASGS uses for GFS-forced runs. What's the best way to instruct ASGS to force ADCIRC with that file?" I don't think there's a need to specifically build a process around MetGet, although that is a convenient way of automating downloads (blocking issues and such aside) so I'll probably start there.

**Reply by wwlwpd** [reply]
I don't think there is a generic way to do this, though it'd be nice to allow for something like:

```
# note: this doesn't actually work, but looks cool! :)
BACKGROUNDMET=CUSTOM
BACKGROUNDMET_FETCH=my-custom-getter-script.pl
```

Effectively, we could make `TROPICALCYCLONE=on` equivalent to `BACKGROUNDMET=NHC`, and in this case would also look for `STORM` and `YEAR`. That's not how it works, but I am thinking is what you're wanting? If so, please create a ticket - not promises, but it's another improvement I think would be good for general use (a lot of code looks at the value of `BACKGROUNDMET` and `TROPICALCYCLONE`.

In the meantime, @jasonfleming is the best person to suggest a work around.
